src/components/Blocks/SearchFarmacia/Body.jsx

```diff
@@ -0,0 +1,383 @@
+        (item) =>
+          item.comune &&
+          item.comune.toLowerCase() === filters.comune.toLowerCase(),
+      );
+    }
+
+    // Ferie - filtro Località
+    if (filters.localita && filters.localita !== null) {
+      newResults = newResults.filter(
+        (item) =>
+          item.localita.toLowerCase() === filters.localita.toLowerCase(),
+      );
+    }
+
+    // set sorting
+    const asc = filters.order.sort_order === 'ascending' ? 1 : -1;
+    newResults = newResults.sort((a, b) =>
+      a[filters.order.sort_on] >= b[filters.order.sort_on] ? asc : -asc,
+    );
+    // set results
+    setResults([...newResults]);
+    // setResults(newResults);
+    setCurrentPage(1);
+  };
+
+  // const doDebouncedFilter = useCallback(
+  //   () => debounce(doFilter, 400),
+  //   [doFilter],
+  // );
+
+  // check if select Comune is empty or not
+  // const checkClearComune = (inputComune) => {
+  //   if (searchFarmacia.items?.length > 0) {
+  //     let resultsByComune = searchFarmacia.items;
+  //     let options = { comuni: [], lista_localita: [] };
+  //     let localitaOptions = [];
+
+  //     if (inputComune === null) {
+  //       localitaOptions = [
+  //         ...new Set(searchFarmacia.items.map((item) => item.localita).sort()),
+  //       ];
+  //       createSelectOptions(localitaOptions, options.lista_localita);
+  //       setFiltersOptions({
+  //         ...filtersOptions,
+  //         lista_localita: options.lista_localita,
+  //       });
+  //     } else {
+  //       resultsByComune = resultsByComune.filter(
+  //         (item) =>
+  //           item.comune.toLowerCase() === inputComune.value.toLowerCase(),
+  //       );
+  //       localitaOptions = [
+  //         ...new Set(resultsByComune.map((item) => item.localita).sort()),
+  //       ];
+  //       createSelectOptions(localitaOptions, options.lista_localita);
+  //       setFiltersOptions({
+  //         ...filtersOptions,
+  //         lista_localita: options.lista_localita,
+  //       });
+  //     }
+  //   }
+  // };
+
+  // function to create object select options
+  // function createSelectOptions(arrayOptions, selectOptions) {
+  //   // eslint-disable-next-line no-unused-expressions
+  //   arrayOptions?.forEach((option) => {
+  //     selectOptions.push({
+  //       value: option.toLowerCase(),
+  //       label: option,
+  //     });
+  //   });
+  // }
+
+  // trigger for filters
+  // useEffect(() => {
+  //   console.log('filters', filters);
+  //   console.log('searchFarmacia', searchFarmacia);
+  //   if (searchFarmacia?.items) {
+  //     doDebouncedFilter(filters, searchFarmacia.items, searchType);
+  //   }
+  // }, [filters]);
+  const doSearch = () => {
+    doFilter(filters, searchFarmacia.items, searchType);
+  };
+  useEffect(() => {
+    searchFarmacia.items && doFilter(filters, searchFarmacia.items, searchType);
+  }, [filters, searchFarmacia.items, searchType]);
+
+  // useDebouncedEffect(
```

Comment:
perchè hai tolto i debounce? 
i debounce servono altrimenti parte subito la chiamata al BE quando un utente digita una sola lettera nel campo di ricerca

Reply:
Testato. Considera che qui non fa chiamate. A livello di uso qui risultava più confortevole senza che con